src/providers/fail_over.c

```diff
@@ -1188,9 +1189,17 @@ fo_resolve_service_done(struct tevent_req *subreq)
                                     &common->rhostent);
     talloc_zfree(subreq);
     if (ret != EOK) {
-        DEBUG(SSSDBG_CRIT_FAILURE, "Failed to resolve server '%s': %s\n",
+        if (resolv_status == EFAULT)
```

Comment:
If we assume here that the error was caused by unable to read `"/etc/hosts"`? If this is only scenario when `resolv_status == EFAULT` ?

Reply:
When server is evaluated two options are possible:
- Querying files - calls `resolv_gethostbyname_files_send()` which lands on `ares_gehostbyname_files()` [1]
- Querying DNS - calls `resolv_gethostbyname_dns_send()` which lands on `ares_search()` [2]

In both cases `ares_*` function return code is stored in `resolv_status`. To be sure that we are dealing with file IO error I thing you can just check for `ARES_EFILE`.

After fast look into `ares_gethostbyname_files()` source code it looks like the only file it is processing is `/etc/hosts` - so you are right. But I am wondering if we should put extra header for it or just hardcode it here - as it is already hardcoded on ares side?

[1] https://c-ares.haxx.se/ares_gethostbyname_file.html
[2] https://c-ares.haxx.se/ares_search.html